fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/recordrepair/RecordValidateOnlyTest.java

```diff
@@ -339,32 +278,47 @@ void testValidateRecordsCorruptVersion(FormatVersion formatVersion, RecordRepair
             storeBuilder = store.asBuilder();
         }
 
-        RecordRepairRunner runner = RecordRepairRunner.builder(fdb).build();
+        RecordRepair.Builder builder = RecordRepair.builder(fdb, storeBuilder).withNumOfRetries(0).withValidationKind(validationKind);
         // We don't currently support detecting this king of error
-        Assertions.assertThatThrownBy(() -> runner.runValidationStats(storeBuilder, validationKind)).hasCauseInstanceOf(UnknownValidationException.class);
-        Assertions.assertThatThrownBy(() -> runner.runValidationAndRepair(storeBuilder, validationKind, false)).hasCauseInstanceOf(UnknownValidationException.class);
-    }
-
-    // list of arguments for version and a bitset that has all the combinations of 4 bits set (except all unset)
-    private static Stream<Arguments> versionAndBitset() {
-        return ParameterizedTestUtils.cartesianProduct(
-                ValidationTestUtils.formatVersions(),
-                ValidationTestUtils.splitsToRemove());
+        try (RecordRepairStatsRunner statsRunner = builder.buildStatsRunner();
+                RecordRepairValidateRunner repairRunner = builder.buildRepairRunner(false)) {
+            RepairStatsResults repairStats = statsRunner.run().join();
+            RepairValidationResults repairResults = repairRunner.run().join();
+            // Iteration stopped short of the full scan
+            Assertions.assertThat(repairStats.getExceptionCaught().getCause()).isInstanceOf(UnknownValidationException.class);
+            Assertions.assertThat(repairStats.getStats())
+                    .hasSize(1)
+                    .containsEntry(RecordRepairResult.CODE_VALID, ValidationTestUtils.RECORD_INDEX_WITH_THREE_SPLITS);
+            Assertions.assertThat(repairResults.isComplete()).isFalse();
+            Assertions.assertThat(repairResults.getCaughtException().getCause()).isInstanceOf(UnknownValidationException.class);
+            Assertions.assertThat(repairResults.getValidResultCount()).isEqualTo(ValidationTestUtils.RECORD_INDEX_WITH_THREE_SPLITS);
+            Assertions.assertThat(repairResults.getInvalidResults()).isEmpty();
+        }
     }
 
     /**
-     * Don't store any versions but verify versions, so there would be many results (all records missing versions).
+     * Delete all versions so there would be many results (all records missing versions).
      * Validate the max number of results.
      * @param maxResultSize the max result size to return
      */
-    @ParameterizedTest
     @CsvSource({"-1", "0", "1", "10", "100"})
```

Comment:
We should probably test with `NUM_RECORDS - 1`

Reply:
done.